### PR TITLE
data: Refactor the Frame struct

### PR DESCRIPTION
**Pull request by Luni-4**
- Move private functions into the `Frame` struct
- Document the `Frame` struct
- Remove same functions

Thanks in advance for your review! :)

### CODE REVIEW

File: data/src/frame.rs [hunk 1]
```diff
@@ -423,106 +419,90 @@ impl DefaultFrameBuffer {
...
-            d_linesize,
-            s,
-            s_linesize,
-            c.unwrap().get_width(w),
-            c.unwrap().get_height(h),
-        );
-    }
-}
+    /// Copies a single video frame in an output buffer.
+    pub fn copy_video_frame<'a, IM, IU>(
```
Comment (lu-zero) [inline data/src/frame.rs:471]: I'm not sure we have to make it pub.

Comment (Luni-4) [reply data/src/frame.rs:471]: Is there a particular reason?

Comment (Luni-4) [reply data/src/frame.rs:471]: A possible solution would be run `copy_video_plane` per the number of planes and save them into n_planes different buffers, then merge all of them into a single one. A bit more cumbersome for the user perhaps, but at least we remove an API :)

Comment (lu-zero) [reply data/src/frame.rs:471]: mediakind-specific API look quite bad on the Frame.

**Pull request merged**